src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -723,11 +723,18 @@ public override BoundNode VisitConversion(BoundConversion node)
             switch (node.ConversionKind)
             {
                 case ConversionKind.MethodGroup:
+                    if (node.Operand is BoundMethodGroup group)
+                    {
+                        checkValidScopedMethodConversion(group.Syntax, node.Conversion.Method, node.Type, invokedAsExtensionMethod: node.IsExtensionMethod, _diagnostics);
```

Comment:
>checkValidScopedMethodConversion(group.Syntax, node.Conversion.Method, node.Type, invokedAsExtensionMethod: node.IsExtensionMethod, _diagnostics); [](http://example.com/codeflow?start=24&length=146)

I would expect this check and the one below to be performed by the Binder, when it creates the BoundConversion node.